### PR TITLE
Add ListObjectsV1 API support

### PR DESCRIPTION
**Pull request by harshavardhana**
This is added for legacy requirements

### CODE REVIEW

**Comment by harshavardhana** [comment]
Not sure why this is failing, but it runs fine locally.

File: docs/API.md [hunk 1]
```diff
@@ -275,6 +276,58 @@ main = do
 
 ```
 
+<a name="listObjectsV1"></a>
+### listObjectsV1 :: Bucket -> Maybe Text -> Bool -> C.Producer Minio ObjectInfo
+
+List objects in the given bucket (version 1)
```
Comment (donatello) [inline docs/API.md]: We should note here that this is an older API and is provided for compatibility. Also, we should mention that `listObjects` implements the V2 API.

Comment (harshavardhana) [reply docs/API.md]: Sure.

**Comment by harshavardhana** [comment]
rebased please review again.. @donatello @krisis

**Pull request merged**